app/uk/gov/hmrc/cdsimportsdds/controllers/DeclarationController.scala

```diff
@@ -38,10 +39,22 @@ class DeclarationController @Inject()(
   extends RESTController(controllerComponents) {
 
   def create(): Action[ImportsDeclarationRequest] = Action.async(parsingJson[ImportsDeclarationRequest]) { implicit request =>
-    val eori = request.headers.get("X-EORI-Identifier")
+    val eori = getEori(request)
     eori match {
       case Some(eori) => saveImportsDeclaration(request.body, eori)
-      case _ => Future.successful(BadRequest(Json.toJson(ErrorResponse("X-EORI-Identifier header missing"))))
+      case _ => missingEoriResponse
+    }
+  }
+
+  private def missingEoriResponse: Future[Result] = {
+    Future.successful(Unauthorized(Json.toJson(ErrorResponse("X-EORI-Identifier header missing"))))
+  }
+
+  def findAll(): Action[AnyContent] = Action.async { implicit request =>
+    val eori = getEori(request)
+    eori match {
```

Comment:
given we have similar pattern matching cases for `eori` from `create`, why not create a separate function and pass  `saveImportsDeclaration` / `findByEori`  to be actioned when needed, just to avoid some repeated lines there.
Something like this maybe ``` def getOrSave(eori: Some(EORI)) (action: EORI => Future[Result]) = { //pattern matching cases }```

Reply:
Should missingEoriResponse be a val?

Reply:
we could, but the source/validation of EORI might be changing when we add auth to these resources, so those pattern matches might go away (or at least be handled in one place in the auth action)

Reply:
have made missingEoriResponse a val for now